src/NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp

```diff
@@ -0,0 +1,104 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Declares functions for interpolating variables to a set of points.
+
+#pragma once
+
+#include <cstddef>
+#include <ostream>
+
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Index;
+namespace PUP {
+class er;
+}  // namespace PUP
+// IWYU pragma: no_forward_declare Variables
+/// \endcond
+
+/// \ingroup NumericalAlgorithmsGroup
+/// Interpolates a `Variables` onto an arbitrary set of points.
+template <size_t Dim>
+class IrregularInterpolant {
+ public:
+  IrregularInterpolant(const Index<Dim>& mesh,
+                       const tnsr::I<DataVector, Dim>& points) noexcept;
+  IrregularInterpolant() = default;
+
+  // clang-tidy: no runtime references
+  /// Serialization for Charm++
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  //@{
+  /// Performs the interpolation on a `Variables` with grid points corresponding
+  /// to the `Index<Dim>` specified in the constructor.
+  /// The result is a `Variables` whose internal `DataVector` goes over the
+  /// list of points that were specified in the constructor.
+  /// \note for the void function, `result` will be resized to the proper size.
+  template <typename TagsList>
+  void interpolate(gsl::not_null<Variables<TagsList>*> result,
+                   const Variables<TagsList>& u) const noexcept;
+  template <typename TagsList>
+  Variables<TagsList> interpolate(const Variables<TagsList>& u) const noexcept;
+  //@}
+
+ private:
+  // clang-tidy: redundant declaration.
+  // (MS: this is needed in order to declare operator== as a friend.)
+  template <size_t ADim>
+  friend bool operator==(const IrregularInterpolant<ADim>& /*lhs*/,  // NOLINT
+                         const IrregularInterpolant<ADim>& /*rhs*/) noexcept;
+  Matrix interpolation_matrix_;
+};
+
+template <size_t Dim>
+template <typename TagsList>
+void IrregularInterpolant<Dim>::interpolate(
+    gsl::not_null<Variables<TagsList>*> result,
+    const Variables<TagsList>& u) const noexcept {
+  // For matrix multiplication of Interp . Source = Result
```

Comment:
I initially missed the context for this first comment line. Could you add a `:` to the end and indent the next three lines 2 additional spaces or "bullet" (`-`) them?